### DISCUSSION TITLE
Trying to figure out the error for "gmake: *** No rule to make target"

### DISCUSSION THREAD

**Discussion by atulkprajapati** [contiki-ng/contiki-ng#1572] [Q&A] [answered]
I have done the setup as per the steps mentioned in the tutorial: https://github.com/contiki-ng/contiki-ng/wiki/Platform-simplelink
I'm too trying to run the examples (hello-world & cc26x0-web-demo), for now I'm getting stuck at build step with:
"gmake: *** No rule to make target 'hello-world.elf' "

Plus the only thing I can see is a cross at project name, no cross on either files to check for the error.
Attached the screenshot for reference. 
Kindly please guide me, to get it resolved. Thank you.

![Screen Shot 2021-05-13 at 8 53 05 PM](https://user-images.githubusercontent.com/22424087/118147468-3aebec00-b42d-11eb-94a7-a9acd2054236.png)

**Comment by atiselsts** [comment, marked as answer]
Unfortunately the Contiki-NG build system has changed significantly since the tutorial was written. You should be able to build with the target equal to `hello-world` (without the `.elf`). If that does not work for some reason, try using the target name as the extension: `hello-world.cc26x0-cc13x0`.

**Reply by PracticalResearch** [reply]
We updated [the wiki page for SimpleLink platform](https://github.com/contiki-ng/contiki-ng/wiki/Platform-simplelink) on November 5, 2021. Based on this update, the debugging should work well. The main problem was resolved by 'TARGET = simplelink'.

**Reply by M-Creeper** [reply]
Hey, I can compile my Contiki-code (Hello World) but all I get is a Hello-World.simplelink file.
Is that file the .elf file I can upload via FlashProgrammer 2, meaning I can simply rename the ending, or did I miss something?

**Comment by atulkprajapati** [comment]
> hello-world.cc26x0-cc13x0

Thank you, it works without an extension and with "hello-world.cc26x0-cc13x0"
But still it's facing the issue with hex conversion.
![Screen Shot 2021-05-14 at 12 11 39 AM](https://user-images.githubusercontent.com/22424087/118171702-21f13400-b449-11eb-9c70-951d04558bc3.png)

**Reply by g-oikonomou** [reply]
Is `srecord` installed in your system and in the path? Try passing `V=1` to the make command for more informative output. 

Also, try pasting console text instead of screenshots - I am sure you should be able to copy from within your IDE. This will make the error possible to index by search engines.

**Reply by atulkprajapati** [reply]
Yeah, I'll make sure of the instructions. 
I checked, it states: srecord is installed.
`srecord 1.64 is already installed and up-to-date.
To reinstall 1.64, run:
  brew reinstall srecord
  
Path variable:
/Library/Frameworks/Python.framework/Versions/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/atulkumarprajapati/Downloads/gcc-arm-none-eabi-9-2019-q4-major/bin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands`

Ya I can see the error as :
$(TRACE_SREC_CAT)
../../arch/cpu/arm/Makefile.arm:81: recipe for target 'build/cc26x0-cc13x0/launchpad/cc2650/hello-world.hex' failed

**Reply by atulkprajapati** [reply]
TRACE_SREC_CAT
On a mac, I'm unable to locate srecord, kindly please guide, how do I sort it out ?

**Reply by g-oikonomou** [reply]
In my setup (also mac), srec_cat is a link under `/usr/local/bin`:

```
$ which srec_cat
/usr/local/bin/srec_cat
$ ls -al /usr/local/bin/srec_cat
lrwxr-xr-x  1 cexgo  admin  35 21 Oct  2014 /usr/local/bin/srec_cat -> ../Cellar/srecord/1.64/bin/srec_cat
```

Normally homebrew will create those symlinks during installation, but you could try to force homebrew to re-link it. `brew help link` should give you a starting point. Or, if you can find srecord under the homebrew installation dir, you can symlink srec_cat in your path manually. Homebrew puts things under `/usr/local/Cellar`. Try this to start with:

```
$ find /usr/local/Cellar/ -name srecord
/usr/local/Cellar//srecord
/usr/local/Cellar//srecord/1.64/include/srecord
/usr/local/Cellar//srecord/1.64/share/doc/srecord
```

Please note: If you create symbolink links manually you may as well link `srec_info`. The build system does not require it, but it's a useful tool to have access to anyway!

**Reply by atulkprajapati** [reply]
I don't quite understand, what it's trying to say. I mean, I can replicate each and everything that you have said above.
So it means that the path is already defined, then I'm unable to understand what I need to do actually.

![Screen Shot 2021-05-18 at 12 13 01 PM](https://user-images.githubusercontent.com/22424087/118604316-a7227300-b7d2-11eb-8ad9-34eb6818cf80.png)

**Reply by alexrayne** [reply]
it try say that  SREC_CAT not executed, ordinary cause it absent.
if you work on windows - it surely absent, and need to find srec for windows. Or just ignore srec generation if srec absent, like in this [patch](https://github.com/alexrayne/contiki-ng/commit/97d46178cd6450e0023bcc019921bbeaf3c90c87)

ordinary srec not need, elf enough.

**Reply by atulkprajapati** [reply]
Well, thanks a lot, although it shows error, but I can direct the debugger to the absolute path of .elf file produced. But although I have taken hello-world project, it's starts debugging from contiki-main.c and stops after 3 step-overs, the debugging comes to a halt. How do I debug the hello-world.c (for that matter any project in examples.)

**Reply by atulkprajapati** [reply]
> it try say that SREC_CAT not executed, ordinary cause it absent.
> if you work on windows - it surely absent, and need to find srec for windows. Or just ignore srec generation if srec absent, like in this [patch](https://github.com/alexrayne/contiki-ng/commit/97d46178cd6450e0023bcc019921bbeaf3c90c87)
> 
> ordinary srec not need, elf enough.


Actually, my ultimate goal is to add an external sensor and read on TI CC2650 Launchpad (running cc2650-web-demo example), the sensor I have is currently DHT11. I have enabled the DIO23 pin and can get the values in mV, but I wanted the actual temperature and humidity values. I have been stuck on it, for a very long time. So, I thought that if I'm able to debug an example from Contiki examples in CCS, it would help in getting the actual values of the sensor. Any guidance in this regard is highly appreciated. Thank you in advance.

**Reply by PracticalResearch** [reply]
I have been debugging some of the examples (including ‘hello-world’) using CC2650STK (https://www.ti.com/tool/CC2650STK) and CC-DEVPACK-DEBUG (https://www.ti.com/tool/CC-DEVPACK-DEBUG) under CCS. If I could know more about the details of your configuration, I might be able to help you to do what you want.

**Reply by atulkprajapati** [reply]
What details do I need to provide ?
I have a CC2650 launchpad as well as CC2650STK, and I'm trying to debug the examples, but after 3 step-overs the program comes to a halt, and it only opens the contiki-main.c and not hello-world.c
Can you please help me with your email-ID and the details that you need, I'll email them to you. Thanks in advance.

**Reply by PracticalResearch** [reply]
I had left a message for you (atulp201) on Gitter (via private chat) which is a good option to exchange the information which might seem unnecessary to be placed on the GitHub’s discussions.

**Comment by PracticalResearch** [comment]
We updated [the wiki page for SimpleLink platform](https://github.com/contiki-ng/contiki-ng/wiki/Platform-simplelink) on November 5, 2021. Based on this update, the debugging should work well. The main problem was resolved by 'TARGET = simplelink'.
